Coming soon: remove flag and permanently enable v2 for site settings

ramonjd opened the pull request:
### Changes proposed in this Pull Request

Coming Soon (public not-indexable by default) has been in production and running smoothly. By "smoothly" we mean no serious bug reports, user difficulties or cataclysmic events.

This PR will be one in a series that removes the coming-soon-v2 flag from Calypso and, eventually, removes the flag from the config files altogether. The consequence is that we are permanently excising v1 (private by default).

This PR removes the coming soon flag check for site settings, thereby removing the v1 private-by-default form fields. All new sites will therefore be able to toggle in an out of Coming Soon v2™️ mode!

The form will display the correct selection for existing Coming Soon v1 sites, that is, if the site is private and unlaunched or set to Coming Soon, the Coming Soon radio button will be checked.

When they toggle Coming Soon mode off and on however, they'll be migrated to Coming Soon v2.

### Testing instructions

Run the unit tests:

` npm run test-client client/my-sites/site-settings/test/form-general.jsx`

Toggle between public and private and coming soon

<img width="764" alt="Screen Shot 2020-11-30 at 12 11 24 pm" src="https://user-images.githubusercontent.com/6458278/100559070-390df080-3305-11eb-8dd0-727e4ddf7aa7.png">

After each toggle check that your site is in the corresponding mode in an incognito window.

ramonjd committed:
Removing coming soon flag check for site settings, thereby removing the v1 private-by-default form fields

p-jackson commented:
@ramonjd The unit test failure is real. Looks like we need to delete all the `blog_public states (coming-soon-v1)` tests in `client/my-sites/site-settings/test/form-general.jsx`. And we can rename the `coming-soon-v2` tests since we don't need to refer to v2 anymore.

ramonjd commented:
> The unit test failure is real. Looks like we need to delete all the blog_public states (coming-soon-v1) tests in client/my-sites/site-settings/test/form-general.jsx.

Argh, good call. Thanks for checking. I think I might add some tests to cover the extra conditions I've thrown in there too, e.g., atomic plugin availability

ramonjd committed:
Unit tests:
- removed references to the coming-soon-v2 flag since we assume it's always enabled
- cover cases where sites are still in coming soon v1 mode
- cover cases where the coming soon etk plugin is disabled (atomic)

andrewserong reviewed (approved):
This is testing well for me, and the e2e test failures look unrelated. 🤞 a rebase today will get them passing, as it looks like the e2es have been working today.

andrewserong commented on client/my-sites/site-settings/test/form-general.jsx:
```diff
@@ -276,6 +229,130 @@ describe( 'SiteSettingsFormGeneral ', () => {
+					const radioButtonComingSoon = getByLabelText( 'Coming soon', { exact: false } );
+					expect( radioButtonComingSoon ).not.toBeChecked();
+
+					// Check that we're not displaying the site as private
```
I think this comment should say: (remove the `not`)
```suggestion
					// Check that we're displaying the site as private
```

andrewserong commented on client/my-sites/site-settings/test/form-general.jsx:
```diff
@@ -276,6 +229,130 @@ describe( 'SiteSettingsFormGeneral ', () => {
+		} );
+
+		describe( 'Coming soon plugin availability', () => {
+			test( 'Should hide Coming Soon form element when the site is not atomic or the editing toolkit plugin is not disabled', () => {
```
Add the word "not" to the test description, because this is testing that the Coming Soon form element is not hidden.

```suggestion
			test( 'Should not hide Coming Soon form element when the site is not atomic or the editing toolkit plugin is not disabled', () => {
```

andrewserong commented:
The unit tests are looking good to me, too, just added a couple of tiny suggested changes to a comment and a test name, nothing critical :)

ramonjd commented:
> The unit tests are looking good to me, too, just added a couple of tiny suggested changes to a comment and a test name, nothing critical :)


Committed those suggestions and rebased. Thank you for being so thorough!
